File: 3555-final-array-state-after-k-multiplication-operations-i/3555-final-array-state-after-k-multiplication-operations-i.java
class Solution {
    public int[] getFinalState(int[] nums, int k, int multiplier) {
        while(k-->0){
            int idx=findMin(nums);
            nums[idx]=nums[idx]*multiplier;
        }
        return nums;
    }

    public int findMin(int[] arr) {
        int min = Integer.MAX_VALUE;
        int findex=-1;
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] < min) {
                min = arr[i];
                findex=i;
            }
        }
     return findex;
    }
}

File: 3555-final-array-state-after-k-multiplication-operations-i/3555-final-array-state-after-k-multiplication-operations-i.py
class Solution:
    def getFinalState(self, nums: List[int], k: int, multiplier: int) -> List[int]:
        for _ in range(k):
            min_idx=0
            for j in range(len(nums)):
                if nums[j] < nums[min_idx] :
                    min_idx=j

            nums[min_idx] *=multiplier

        return nums

        